test/views/MaterialsAmountViewSpec.scala

```diff
@@ -0,0 +1,231 @@
+        implicit val document: Document = Jsoup.parse(underTest(pageModel).body)
+
+        welshToggleCheck(userScenario.isWelsh)
+        titleCheck(userScenario.specificExpectedResults.get.expectedTitle)
+        captionCheck(userScenario.commonExpectedResults.expectedCaption(taxYearEOY))
+        h1Check(userScenario.specificExpectedResults.get.expectedH1("some-contractor"))
+        textOnPageCheck(userScenario.specificExpectedResults.get.expectedReplayContent1("£333.33"), Selectors.paragraphTextSelector(number = 1))
+        textOnPageCheck(userScenario.specificExpectedResults.get.expectedReplayContent2(translatedMonthAndTaxYear(pageModel.month, taxYearEOY)), Selectors.paragraphTextSelector(number = 2))
+        textOnPageCheck(userScenario.specificExpectedResults.get.expectedOnlyIncludeVATParagraph, Selectors.paragraphTextSelector(number = 3))
+        hintTextCheck(userScenario.commonExpectedResults.expectedHintText, Selectors.hintTextSelector)
+        textOnPageCheck(text = "£", Selectors.poundPrefixSelector)
+        inputFieldValueCheck(AmountForm.amount, Selectors.inputFieldSelector, value = "333.33")
+        formPostLinkCheck(MaterialsAmountController.submit(taxYearEOY, Month.JUNE.toString.toLowerCase, contractor = "some-ref").url, Selectors.formSelector)
+        buttonCheck(userScenario.commonExpectedResults.expectedButtonText, Selectors.buttonSelector)
+      }
+
+      "render page with form containing empty form error" which {
+        implicit val userSessionDataRequest: UserSessionDataRequest[AnyContent] = getUserSessionDataRequest(userScenario.isAgent)
+        implicit val messages: Messages = getMessages(userScenario.isWelsh)
+
+        val form = new FormsProvider().materialsAmountForm(isAgent = userScenario.isAgent).bind(Map(AmountForm.amount -> ""))
+        val pageModel = aMaterialsAmountPage.copy(contractorName = Some("some-contractor"), employerRef = "some-ref", form = form)
+        implicit val document: Document = Jsoup.parse(underTest(pageModel).body)
+
+        welshToggleCheck(userScenario.isWelsh)
+        titleCheck(userScenario.specificExpectedResults.get.expectedErrorTitle)
+        captionCheck(userScenario.commonExpectedResults.expectedCaption(taxYearEOY))
+        h1Check(userScenario.specificExpectedResults.get.expectedH1("some-contractor"))
+        textOnPageCheck(userScenario.specificExpectedResults.get.expectedTellUsTheAmountText(translatedMonthAndTaxYear(pageModel.month, taxYearEOY)), Selectors.paragraphTextSelector(number = 2))
+        textOnPageCheck(userScenario.specificExpectedResults.get.expectedOnlyIncludeVATParagraph, Selectors.paragraphTextSelector(number = 3))
+        hintTextCheck(userScenario.commonExpectedResults.expectedHintText, Selectors.hintTextSelector)
+        textOnPageCheck(text = "£", Selectors.poundPrefixSelector)
+        errorSummaryCheck(userScenario.specificExpectedResults.get.expectedErrorNoEntry, Selectors.expectedErrorHref)
```

Comment:
we're missing the check for the error message that appears above the amount field box (errorAboveElementCheck). It looks like this has been missed in our other view specs too. Is this something we want to add now or in a follow-up/refactoring PR?

Reply:
as discussed this is part of a different element and does not need to be tested here